onnxruntime/test/contrib_ops/ngram_test.cc

```diff
@@ -0,0 +1,565 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "gtest/gtest.h"
+#include "test/providers/provider_test_utils.h"
+
+#include <stdint.h>
+
+namespace onnxruntime {
+namespace test {
+namespace ngram_test {
+
+constexpr const char* domain = onnxruntime::kMSDomain;
+const int opset_ver = 1;
+
+void InitTestAttr(OpTester& test, const std::string& mode,
+                  int64_t min_gram_length, int64_t max_gram_length, int64_t max_skip_count,
+                  const std::vector<int64_t>& ngram_counts,
+                  const std::vector<int64_t>& ngram_indexes,
+                  const std::vector<float>& weights,
+                  const std::vector<int64_t>& pool_int64s,
+                  const std::vector<std::string>& pool_strings) {
+  test.AddAttribute("mode", mode);
+  test.AddAttribute("min_gram_length", min_gram_length);
+  test.AddAttribute("max_gram_length", max_gram_length);
+  test.AddAttribute("max_skip_count", max_skip_count);
+  test.AddAttribute("ngram_counts", ngram_counts);
+  test.AddAttribute("ngram_indexes", ngram_indexes);
+  // optional
+  if (!weights.empty()) {
+    test.AddAttribute("weights", weights);
+  }
+  if (!pool_int64s.empty()) {
+    test.AddAttribute("pool_int64s", pool_int64s);
+  } else {
+    test.AddAttribute("pool_strings", pool_strings);
+  }
+}
+}  // namespace ngram_test
+
+using namespace ngram_test;
+
+// Here is what takes place in general and in particular
+// in this unit test.There are 7 n - grams : 4 unigrams and 3 bigrams
+// that are expressed as 10 items(integers in this case) contained within pool_int64 attribute.
+// We only count and then optionally scale those ngrams that appear in the supplied pool parameter(either int64 or string).
+// M = 1 and N = 2 in this case.
+// However, attribute all controls whether we consider all of the supplied ngram[M..N] sizes
+// into consideration or not.With all = false, we only consider N - grams.
+
+TEST(ContribOpTest, Ngram_Int32_TF_onlyBigrams_Skip0) {
```

Comment:
>ContribOpTest [](start = 5, length = 13)

Should we just use the op name (NgramTest), or at least include it in the test suite name (ContribOpNgramTest) so we separate out the tests for the different contrib ops a little more?